### PR TITLE
Don't send ciphersuites twice in QUIC

### PR DESCRIPTION
**Pull request by mattcaswell**
QUIC TLS was sending some ciphersuites twice in the ClientHello. This
was due to us declaring some TLSv1.3 ciphersuites in the list intended to
describe the TLSv1.2 ciphersuites supported by the SSL_METHOD.

### CODE REVIEW

**Comment by t8m** [comment]
CI is relevant

**Comment by mattcaswell** [comment]
> CI is relevant

Fixup pushed.

File: ssl/ssl_ciph.c [hunk 1]
```diff
@@ -1495,147 +1495,149 @@ STACK_OF(SSL_CIPHER) *ssl_create_cipher_list(SSL_CTX *ctx,
...
+                                            &head, &tail, ca_list, c);
+            rule_p += 7;
+            if (*rule_p == ':')
+                rule_p++;
+        }
 
-    if (ok && (rule_p[0] != '\0'))
-        ok = ssl_cipher_process_rulestr(rule_p, &head, &tail, ca_list, c);
+        if (ok && (rule_p[0] != '\0'))
+            ok = ssl_cipher_process_rulestr(rule_p, &head, &tail, ca_list, c);
```
Comment (t8m) [inline ssl/ssl_ciph.c:1633]: I believe this means the `@SECLEVEL=x` in the cipher string won't be processed for QUIC.

**Review by t8m — CHANGES_REQUESTED** [review]

File: ssl/ssl_ciph.c (hunk 1 above)
Comment (mattcaswell) [reply ssl/ssl_ciph.c]: Fixed

**Comment by mattcaswell** [comment]
All feedback addressed.

Ping for reviews.

**Review by paulidale — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.

**Pull request closed without merging**